### PR TITLE
[codex] Add Qwen36 packed FFN diagnostics and SiLU drift lane

### PR DESCRIPTION
**Pull request by DeanoC**
## Summary
- Add Qwen3.6 packed/static routed FFN diagnostics, including static Top-N partial probing and packed-family parity reporting.
- Record the recent Qwen3.6 router/scheduling/barrier probe data in `docs/detailed_performance.md`.
- Add exact routed FFN diagnostic coverage for host-order gate/up and routed down on Metal.
- Add an actual Metal SiLU tap so packed FFN drift can be attributed to routed expert SiLU ulp differences instead of inferred from CPU recomputation.
- Add named sweep/report modes for exact host-order routed FFN and known SiLU ULP drift classification.

## Why
The packed Qwen3.6 routed FFN path can match short strict decodes but diverges later. The investigation pinned the first internal drift to actual Metal SiLU evaluation: gate/up dot products match, but Metal SiLU can differ by ~1 ulp, which can cross a BF16 boundary in routed down/finalize.

This PR keeps the diagnostic proof paths separate from promotion/perf paths: exact host-order kernels are intentionally slow and should not be promoted as a fast path, but they make the remaining drift source reproducible.

## Validation
- `cargo build --release -p runner --bin supersonic`
- `PYTHONPYCACHEPREFIX=target/pycache python3 -m py_compile tests/metal/sweep_qwen36_fused_routed_int4.py tests/metal/sweep_qwen36_static_topn_runtime.py`
- 4-token static exact-host-order smoke: `generated_ids_match=true`, known drift status `strict_match`.
- 128-token static exact-host-order diagnostic: `generated_ids_match=false`, classified as `known_long_run_drift` with `first_mismatch_index=74` and `strict_violation_count=0`.

## Notes
This PR includes the local diagnostic stack currently ahead of `main` on `codex/router-logits-exact-pair` (6 commits). The next perf target should stay on the fast packed routed FFN path using this source-aware tolerance/reporting lane.

### CODE REVIEW

File: tests/metal/sweep_qwen36_static_topn_runtime.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 
 MODEL = "qwen3.6-35b-a3b"
-SCHEMA = "qwen36-static-topn-runtime-sweep-v4"
+SCHEMA = "qwen36-static-topn-runtime-sweep-v6"
```
[the remark this answers is not in the document]
Comment (DeanoC) [reply tests/metal/sweep_qwen36_static_topn_runtime.py:18]: Fixed in `bc92599`: updated the SOTA gate and contract test expected schema to `qwen36-static-topn-runtime-sweep-v6`.

Validation: `python3 -m unittest tests.test_qwen36_sota_contracts`.

File: tests/metal/sweep_qwen36_fused_routed_int4.py [hunk 2]
```diff
@@ -16,14 +16,16 @@
 
 
 MODEL = "qwen3.6-35b-a3b"
-SCHEMA = "qwen36-fused-routed-int4-sweep-v42"
+SCHEMA = "qwen36-fused-routed-int4-sweep-v43"
```
[the remark this answers is not in the document]
Comment (DeanoC) [reply tests/metal/sweep_qwen36_fused_routed_int4.py:19]: Fixed in `bc92599`: updated the SOTA gate and contract test expected schema to `qwen36-fused-routed-int4-sweep-v43`.

Validation: `python3 -m unittest tests.test_qwen36_sota_contracts`.

File: crates/kernel-ffi/src/metal_native.mm [hunk 3]
```diff
@@ -17507,17 +17659,19 @@ static int supersonic_metal_l2norm_impl(
...
-                    ? "qwen36_ffn_int4_expert_down_finalize_topk_parallel"
-                    : (expert_down_rowpair_topk_parallel
-                        ? "qwen36_ffn_int4_expert_down_finalize_rowpair_topk_parallel"
-                        : (expert_down_gathered
-                            ? "qwen36_ffn_int4_expert_down_gathered"
-                            : (raw_ggml_experts
-                                ? "qwen36_ffn_int4_expert_down_finalize_multirow"
-                                : "qwen36_ffn_int4_expert_down_finalize"))));
+            const std::string expert_down_label = qwen36_ffn_expert_down_host_order_enabled()
+                ? "qwen36_ffn_int4_expert_down_finalize_host_order_stage5"
```
[the remark this answers is not in the document]
Comment (DeanoC) [reply crates/kernel-ffi/src/metal_native.mm]: Fixed in `bc92599`: the `stage5_with_router` encoder now captures `expert_down_host_order`, checks for `expert_down_finalize_host_order`, selects that pipeline when enabled, and dispatches it with the host-order shape.

Validation: `cargo build --release -p runner --bin supersonic`.

**Comment by DeanoC** [comment]
@codex please re-review
